src/lib/confmgt/unitparse.c

```diff
@@ -142,10 +144,33 @@ config_parse_units(const char *val, const unit_table_t *u, int *ok)
 
   for ( ;u->unit;++u) {
     if (!strcasecmp(u->unit, cp)) {
-      if (use_float)
-        v = (uint64_t)(u->multiplier * d);
-      else
-        v *= u->multiplier;
+      if (use_float) {
+        d = u->multiplier * d;
+
+        if (d < 0) {
+          log_warn(LD_CONFIG, "Negative value arised when parsing %s %s",
+                   val, u->unit);
```

Comment:
We also need to fail on negative values:
```suggestion
                   val, u->unit);
          *ok = 0;
           goto done;  
```

Reply:
Ouch! Sorry for missing this.

I will also add a negative double test so everything is covered.